### PR TITLE
Makes a fake all-access card dispenser

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
Makes a fake captains card similar to the fake emag, now in the arcade
Makes a machine to dispense them

## Why It's Good For The Game
The visual joke of having a *literal* aa dispenser, giving people a mini heart attack when they see an all access on the ground.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

allaccess_ground

allaccess_exam

</details>

## Changelog
:cl:
add: Added the toy captains card
add: Adds the all access dispenser
add: Adds an admin-only version that actually creates captains spare ids
tweak: Adds the card to the arcade prizes
tweak: Adds the dispenser to maint spawns
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Should display as captain if worn in job slot :tf:

**Comment by etherware-novice** [comment]
> Should display as captain if worn in job slot :trollface:

Evil!

**Comment by bluezorua** [comment]
I think the machine should be a maint room rather than random maint loot tbh

File: code/_globalvars/lists/maintenance_loot.dm [hunk 1]
```diff
@@ -111,5 +111,6 @@ GLOBAL_LIST_INIT(maintenance_loot, list(
 	/obj/item/reagent_containers/food/drinks/bottle/homemaderum = 1,
 	/obj/item/reagent_containers/food/snacks/grown/poppy = 10,
 	/obj/item/throwing_star = 1,
+	/obj/machinery/allaccess = 2
```
Comment (PowerfulBacon) [inline code/_globalvars/lists/maintenance_loot.dm]: Spawn the circuit instead of the machine

Comment (Sarchutar) [reply code/_globalvars/lists/maintenance_loot.dm]: also lacks a comma at the end

File: code/game/machinery/allaccess.dm [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/obj/machinery/allaccess
+	name = "All Access Dispenser"
+	desc = "Some people confuse the Head of Personell with this. Contains fake all access cards to keep the greytide at bay!"
```
Comment (itsmeow) [inline code/game/machinery/allaccess.dm]: 'personnel' not 'personell'

File: code/game/machinery/allaccess.dm [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	else
+		if(powered())
+			icon_state = initial(icon_state)
+			stat &= ~NOPOWER
+		else
+			icon_state = "[initial(icon_state)]-off"
+			stat |= NOPOWER
+
+/obj/machinery/allaccess/real
+	desc = "Some people confuse the Head of Personell with this. Contains real All Access!"  // admin spawn for funnies
```
Comment (itsmeow) [inline code/game/machinery/allaccess.dm]: here as well

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -1525,6 +1525,20 @@
 	for(var/i in 1 to rand(1,4))
 		new /obj/item/toy/reality_pierce(src)
 
+/*
+ * Fake captains card
+ */
+
+/obj/item/toy/allaccess
+	name = "captains spare ID"
+	desc = "A replica of the glorious captains card. Issued to annoying greytide as a joke."
```
Comment (itsmeow) [inline code/game/objects/items/toys.dm]: "captain's" card, not "captains card", possibly change "greytide" to "greytiders"

File: code/_globalvars/lists/maintenance_loot.dm (hunk 1 above)
Comment (etherware-novice) [reply code/_globalvars/lists/maintenance_loot.dm]: thats probably why it didnt compile

File: code/game/machinery/allaccess.dm [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+/obj/machinery/vending/power_change()
+	if(stat & BROKEN)
+		icon_state = "[initial(icon_state)]-broken"
+	else
+		if(powered())
+			icon_state = initial(icon_state)
+			stat &= ~NOPOWER
+		else
+			icon_state = "[initial(icon_state)]-off"
+			stat |= NOPOWER
```
Comment (PowerfulBacon) [inline code/game/machinery/allaccess.dm]: Incorrect typepath

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/game/machinery/allaccess.dm (hunk 5 above)
Comment (etherware-novice) [reply code/game/machinery/allaccess.dm]: whats typepath help

File: code/game/machinery/allaccess.dm [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	density = FALSE
+	circuit = /obj/item/circuitboard/machine/allaccess
+	pass_flags = PASSTABLE
+
+/obj/machinery/allaccess/attack_hand(mob/living/user)
+	var/output = new spawnitem
+	user.put_in_active_hand(output)
+	to_chat(user, "<span class='notice'>You take the card out of the dispenser.")
+
+/obj/machinery/vending/power_change()
```
Comment (itsmeow) [inline code/game/machinery/allaccess.dm]: This is a typepath. It's for the vending machine not the aa dispenser

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/game/objects/items/toys.dm [hunk 7]
```diff
@@ -1525,6 +1525,20 @@
 	for(var/i in 1 to rand(1,4))
 		new /obj/item/toy/reality_pierce(src)
 
+/*
+ * Fake captains card
+ */
+
+/obj/item/toy/allaccess
+	name = "captains spare ID"
```
Comment (itsmeow) [inline code/game/objects/items/toys.dm]: missed one

**Review by itsmeow — APPROVED** [review]

File: code/game/machinery/allaccess.dm [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+	var/spawnitem = /obj/item/toy/allaccess
+	idle_power_usage = 5
+	density = FALSE
+	circuit = /obj/item/circuitboard/machine/allaccess
+	pass_flags = PASSTABLE
+
+/obj/machinery/allaccess/attack_hand(mob/living/user)
+	var/output = new spawnitem
+	user.put_in_active_hand(output)
+	to_chat(user, "<span class='notice'>You take the card out of the dispenser.")
```
Comment (itsmeow) [inline code/game/machinery/allaccess.dm]: missing `</span>`

File: code/game/objects/items/toys.dm [hunk 9]
```diff
@@ -1525,6 +1526,20 @@
...
+
+/obj/item/toy/allaccess
+	name = "captain's spare ID"
+	desc = "A replica of the glorious captain's card. Issued to annoying greytiders as a joke."
+	icon = 'icons/obj/card.dmi'
+	icon_state = "gold"
+
+/obj/item/toy/allaccess/afterattack()
+	. = ..()
+	playsound(src, 'sound/items/bikehorn.ogg', 50, 1)
```
Comment (itsmeow) [inline code/game/objects/items/toys.dm]: change the 1 to `TRUE` defines

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Review by itsmeow — APPROVED** [review]

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**